HiAnalysis/HiOnia/plugins/HiOniaAnalyzer.cc

```diff
@@ -799,451 +864,696 @@ HiOniaAnalyzer::fillTreeMuon(const pat::Muon* muon, int iType, ULong64_t trigBit
     return;
   }
 
-  Reco_mu_charge[Reco_mu_size] = muon->charge();
-  Reco_mu_type[Reco_mu_size] = iType;
+  if (muon!=NULL){
+    Reco_mu_charge[Reco_mu_size] = muon->charge();
+    Reco_mu_type[Reco_mu_size] = iType;
   
-  TLorentzVector vMuon = lorentzMomentum(muon->p4());
-  new((*Reco_mu_4mom)[Reco_mu_size])TLorentzVector(vMuon);
+    TLorentzVector vMuon = lorentzMomentum(muon->p4());
+    new((*Reco_mu_4mom)[Reco_mu_size])TLorentzVector(vMuon);
 
-  Reco_mu_trig[Reco_mu_size] = trigBits;
+    Reco_mu_trig[Reco_mu_size] = trigBits;
 
-  reco::TrackRef iTrack = muon->innerTrack();
+    reco::TrackRef iTrack = muon->innerTrack();
   
-  if (!_theMinimumFlag) {
-    Reco_mu_SelectionType[Reco_mu_size] = muonIDmask(muon);
-    Reco_mu_highPurity[Reco_mu_size] = iTrack->quality(reco::TrackBase::highPurity);
-    Reco_mu_isGoodMuon[Reco_mu_size] = muon::isGoodMuon(*muon, muon::TMOneStationTight);
-    Reco_mu_TrkMuArb[Reco_mu_size] = muon->muonID("TrackerMuonArbitrated");
-    Reco_mu_TMOneStaTight[Reco_mu_size] = muon->muonID("TMOneStationTight");
-    Reco_mu_nTrkHits[Reco_mu_size] = iTrack->found();
-    Reco_mu_normChi2_inner[Reco_mu_size] = iTrack->normalizedChi2();
-    Reco_mu_nPixValHits[Reco_mu_size] = iTrack->hitPattern().numberOfValidPixelHits();
-    Reco_mu_nPixWMea[Reco_mu_size] = iTrack->hitPattern().pixelLayersWithMeasurement();
-    Reco_mu_nTrkWMea[Reco_mu_size] = iTrack->hitPattern().trackerLayersWithMeasurement();
-    Reco_mu_StationsMatched[Reco_mu_size] = muon->numberOfMatchedStations();
-    Reco_mu_dxy[Reco_mu_size] = iTrack->dxy(RefVtx);
-    Reco_mu_dxyErr[Reco_mu_size] = iTrack->dxyError();
-    Reco_mu_dz[Reco_mu_size] = iTrack->dz(RefVtx);
-    Reco_mu_dzErr[Reco_mu_size] = iTrack->dzError();
-    Reco_mu_pt_inner[Reco_mu_size] = iTrack->pt();
-    Reco_mu_ptErr_inner[Reco_mu_size] = iTrack->ptError();
+    if (!_theMinimumFlag) {
+      if (!iTrack.isNull()){
+	Reco_mu_SelectionType[Reco_mu_size] = muonIDmask(muon);
+	Reco_mu_highPurity[Reco_mu_size] = iTrack->quality(reco::TrackBase::highPurity);
+	Reco_mu_isGoodMuon[Reco_mu_size] = muon::isGoodMuon(*muon, muon::TMOneStationTight);
+	Reco_mu_TrkMuArb[Reco_mu_size] = muon->muonID("TrackerMuonArbitrated");
+	Reco_mu_TMOneStaTight[Reco_mu_size] = muon->muonID("TMOneStationTight");
```

Comment:
True. I will comment out all references of isGoodMuon variable then (only commenting, because it could be useful in the future, if we want to do isGoodMuon(other_ID_requirement) ).

Reply:
Since we also have the branch called "Reco_mu_SelectionType" which encodes the result of all IDs used in the isGoodMuon function (see HiOniaAnalyzer::muonIDmask function), it would be better to simply remove all instances of the branch Reco_mu_isGoodMuon.